src/viam/sdk/services/navigation.hpp

```diff
@@ -0,0 +1,101 @@
+/// @file services/navigation.hpp
+///
+/// @brief Defines a `Navigation` service.
+#pragma once
+
+#include <string>
+
+#include <viam/api/common/v1/common.pb.h>
+#include <viam/api/service/navigation/v1/navigation.pb.h>
+
+#include <viam/sdk/common/pose.hpp>
+#include <viam/sdk/common/utils.hpp>
+#include <viam/sdk/services/service.hpp>
+#include <viam/sdk/spatialmath/geometry.hpp>
+
+namespace viam {
+namespace sdk {
+
+class Navigation : public Service {
+   public:
+    enum class Mode : uint8_t {
+        k_unspecified,
+        k_manual,
+        k_waypoint,
+        k_explore,
+    };
+
+    enum class MapType : uint8_t {
+        k_unspecified,
+        k_none,
+        k_gps,
+    };
+
+    struct LocationResponse {
+        geo_point location;
+        double compass_heading;
+    };
+
+    struct Waypoint {
+        Waypoint(const viam::service::navigation::v1::Waypoint& proto)
+            : id(proto.id()), location(geo_point::from_proto(proto.location())) {}
+
+        operator viam::service::navigation::v1::Waypoint() {
+            viam::service::navigation::v1::Waypoint ret;
+            *ret.mutable_id() = id;
+            *ret.mutable_location() = location.to_proto();
+            return ret;
+        }
+
+        std::string id;
+        geo_point location;
+    };
+
+    struct Path {
+        Path(const viam::service::navigation::v1::Path& proto)
+            : destination_waypoint_id(proto.destination_waypoint_id()) {
+            repeatedPtrToVec(proto.geopoints(), geopoints);
+        }
+
+        operator viam::service::navigation::v1::Path() {
+            viam::service::navigation::v1::Path ret;
+            *ret.mutable_destination_waypoint_id() = destination_waypoint_id;
+            vecToRepeatedPtr(geopoints, *ret.mutable_geopoints());
+            return ret;
+        }
+
+        std::string destination_waypoint_id;
+        std::vector<geo_point> geopoints;
+    };
+
+    API api() const override;
+
+    virtual Mode get_mode(const std::string name, const ProtoStruct& extra) = 0;
```

Comment:
this header looks good overall, maybe you were already going to do this but we'll want to have the overrides which don't take the `extra` parameter similar to how it looks in the motion service for example

Reply:
added overrides without `extra` in 3f7c6c7f